src/libraries/System.Private.CoreLib/src/System/Random.PrngCompatibility.cs

```diff
@@ -14,11 +14,11 @@ public partial class Random
         /// Provides an implementation used for compatibility with cases where a seed is specified
         /// and thus the sequence produced historically could have been relied upon.
         /// </summary>
-        private sealed class Net5CompatSeedImpl : ImplBase
+        private sealed class SeededRandomImpl : ImplBase
```

Comment:
I would just drop `Net5` from the names: `CompatSeedImpl`, `CompatDerivedImpl`, etc. to make it clear that the primary purpose of these types is compatibility.